### PR TITLE
Clang: Create kits for bundled LLVM in MSVC

### PR DESCRIPTION
**Pull request by Zingam**
## This change addresses item https://github.com/microsoft/vscode-cmake-tools/issues/914

The following changes are proposed:

- Refactor the variables according to the coding standard described in development.rst.
- Enables the detection of LLVM which is bundled in VS2019.

## The purpose of this change

- Detect and create kits for LLVM which is bundled in VS2019.

## Other Notes/Information

This PR basically replaces: #894

### CODE REVIEW

File: src/kit.ts [hunk 1]
```diff
@@ -806,35 +806,61 @@ export async function scanForKits(opt?: KitScanOptions) {
...
+      scan_paths.forEach(path_ => clang_cl_paths.add(path_));
+      // LLVM bundled in VS locations
+      const vs_installs = await vsInstallations();
+      const bundled_clang_cl_paths = vs_installs.map(vs_install => {
+        return vs_install.installationPath + "\\VC\\Tools\\Llvm\\bin";
+      });
+      bundled_clang_cl_paths.forEach(path_ => {clang_cl_paths.add(path_);});
+
+      // Scan for kits
+      const vs_kits = scanForVSKits(pr);
```
Comment (bobbrow) [inline src/kit.ts]: What does changing the scanning order do?  `scanForVSKits` used to be first.  Is there a reason the order is changing?

File: src/kit.ts [hunk 2]
```diff
@@ -806,35 +806,61 @@ export async function scanForKits(opt?: KitScanOptions) {
...
-    let scanPaths: string[] = [];
+    let scan_paths = [] as string[];
+
     // Search directories on `PATH` for compiler binaries
-    const pathvar = process.env['PATH']!;
-    if (pathvar) {
+    if (process.env.hasOwnProperty('PATH')) {
       const sep = isWin32 ? ';' : ':';
-      scanPaths = scanPaths.concat(pathvar.split(sep));
+      scan_paths = scan_paths.concat((process.env.PATH as string).split(sep));
```
[the remark this answers is not in the document]
Comment (Zingam) [reply src/kit.ts]: This doc https://github.com/microsoft/vscode-cmake-tools/blob/develop/docs/development.rst#style states:

Style
- Use inline field initializers whenever possible.
- Declare properties in constructor parameters lists, when possible.
- Use lowerCamelCase for public members, methods, and function/method parameters.
- Use snake_case for variables.
- Use kebab-case for files and directories. (hyphen-separated-names)
- Prefix private members/methods with an underscore and write them _withCamelCase

Most of the variables in the code use the snake_case. So if from now on camel case is the norm, please, fix this document. I actually prefer camelCase anyway.

File: src/kit.ts (hunk 1 above)
Comment (Zingam) [reply src/kit.ts]: I just grouped the scanFor... functions together as in that way it is easier for me to follow the process flow. It doesn't change the extension's behavior.

File: src/kit.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (bobbrow) [reply src/kit.ts]: I didn't see that.  We were planning on aligning this project with cpptools once we closed out the existing PRs, but haven't gotten there yet.  Sorry for the confusion.  I guess that means you can leave the underscores for now, but we will be changing them soon anyway.

[the remark this answers is not in the document]
Comment (Zingam) [reply src/kit.ts]: Thank you for letting me know. I couldn't find a document describing any coding rules for cpptools. So I'll have to look at the code and try to keep inline in the future.
I'll fix the variable names. In this regard I'd like to ask: should I also fix all the variable names in the functions I make changes that use `snake_case` historically?

I already have another branch ready for a Pull Request. It is based on this one #930, which implements #823 with significantly more changes.

[the remark this answers is not in the document]
Comment (bobbrow) [reply src/kit.ts]: I would rather make coding convention changes as a separate PR.  I would just say not to worry about it for now.  You can continue to follow vector-of-bool's guide until we write up a replacement document and make the style changes in the code.  Sorry again for the confusion here.

**Pull request merged**